app/middleware.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from django.conf import settings
+from django.http import HttpRequest
+from django.utils.deprecation import MiddlewareMixin
+from django.utils.functional import SimpleLazyObject
+from praw import Reddit
+from social_django.utils import load_strategy
+
+from blossom import __version__
+
+log = logging.getLogger(__name__)
+
+
+def refresh_token(request: HttpRequest) -> None:
+    """Refresh the OAuth token so that we can use it."""
+    strategy = load_strategy(request)
+    social = request.user.social_auth.filter(provider="reddit")[0]
+    social.refresh_token(
+        strategy=strategy, redirect_uri="http://localhost:8000/complete/reddit/"
```

Comment:
This needs to update depending on what environment we're running, prod vs anything else